elyra/pipeline/airflow/component_parser_airflow.py

```diff
@@ -127,16 +127,25 @@ def _parse_properties(self, content: Dict[str, List]) -> List[ComponentParameter
                 self.log.warning(f"Data type from parsed data ('{data_type}') could not be determined. "
                                  f"Proceeding as if 'string' was detected.")
 
-            properties.append(
-                ComponentParameter(
-                    id=arg,
-                    name=arg,
-                    data_type=data_type_info.data_type,
-                    value=(value or data_type_info.default_value),
-                    description=description,
-                    control_id=data_type_info.control_id
-                )
-            )
+            # Override control id since all input properties can take in an xcom
+            control_id = "OneOfControl"
```

Comment:
Could you please expand on the comment of what "OneOfControl" does or implies?  Based on our scrum conversations, `OneOfControl` is kind of a _meta-control_ that "encapsulates" other controls - one of which is chosen by the user (and thus its name: "OneOfControl").  I believe it's the "NestedEnumControl", unconditionally added below, which adds the "xcom" support while the "default_control_type" is more for handling the native type's input.  (If my understanding is correct.)

Reply:
> `OneOfControl` is kind of a _meta-control_ that "encapsulates" other controls - one of which is chosen by the user (and thus its name: "OneOfControl"). I believe it's the "NestedEnumControl", unconditionally added below, which adds the "xcom" support while the "default_control_type" is more for handling the native type's input. (If my understanding is correct.)

Yes! this is pretty much the gist of it.